compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2843,6 +2858,94 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// Try to match an implementation of `Index` against a self type, and report
+    /// the unsatisfied predicates that result from confirming this impl.
+    ///
+    /// Given an index expression, sometimes the `Self` type shallowly but does not
+    /// deeply satisfy an impl predicate. Instead of simply saying that the type
+    /// does not support being indexed, we want to point out exactly what nested
+    /// predicates cause this to be, so that the user can add them to fix their code.
+    fn find_and_report_unsatisfied_index_impl(
+        &self,
+        index_expr_hir_id: HirId,
+        base_expr: &hir::Expr<'_>,
+        base_ty: Ty<'tcx>,
+    ) -> Option<(ErrorGuaranteed, Ty<'tcx>, Ty<'tcx>)> {
+        let Some(index_trait_def_id) = self.tcx.lang_items().index_trait() else {
+            return None;
+        };
+
+        self.tcx.find_map_relevant_impl(
```

Comment:
Doesn't @lcnr want to remove that method?

Reply:
I could use `for_each_relevant_impl` instead.